frontend/packages/kubevirt-plugin/src/components/vms/vm-resource.tsx

```diff
@@ -156,13 +148,13 @@ export const VMDetailsList: React.FC<VMResourceListProps> = ({
 
       <VMDetailsItem
         title="CD-ROMs"
-        canEdit={canUpdateVM}
+        canEdit={!!vm && canUpdateVM}
         editButtonId={prefixedID(id, 'cdrom-edit')}
         onEditClick={() => VMCDRomModal({ vmLikeEntity: vm, modalClassName: 'modal-lg' })}
         idValue={prefixedID(id, 'cdrom')}
         isNotAvail={cds.length === 0}
       >
-        <DiskSummary disks={cds} vm={vm} />
+        {vm && <DiskSummary disks={cds} vm={vm} />}
       </VMDetailsItem>
 
       <VMDetailsItem
```

Comment:
It will be nice to know if we are in vm or vmi, but we can do it in a follow up because it will need to be tested and we may introduce new bugs if we do it too quickly.

Reply:
editing still possible